src/pairselection.hxx

```diff
@@ -48,15 +78,159 @@ auto buildgenpair(auto &df, const std::string &recopair,
+                Logger::get("buildtruegenpair")
+                    ->debug("|    Pdgid: {}", gen_candidate_1.pdgid);
+                Logger::get("buildtruegenpair")
+                    ->debug("|    motherid: {}", gen_candidate_1.motherid);
+                Logger::get("buildtruegenpair")
+                    ->debug("|    found_correct_mother: {}", found);
+                Logger::get("buildtruegenpair")
+                    ->debug("|    motherPdgid: {}", mother_pdgid);
+                Logger::get("buildtruegenpair")
+                    ->debug("|--------------------------------------------");
+                if (found && genpair[0] == -1) {
+                    genpair[0] = gen_candidate_1.index;
+                } else if (found && genpair[1] == -1) {
+                    genpair[1] = gen_candidate_1.index;
+                }
+            }
+        } else {
+            // in the case of two different daughter particles, we need find
+            // each one seperately
+            for (const auto &gen_candidate_1 : gen_candidates_1) {
+                if (check_mother(genparticles, gen_candidate_1.index,
+                                 mother_pdgid)) {
+                    genpair[0] = gen_candidate_1.index;
+                    break;
+                }
+            }
+            for (const auto &gen_candidate_2 : gen_candidates_1) {
+                if (check_mother(genparticles, gen_candidate_2.index,
+                                 mother_pdgid)) {
+                    genpair[1] = gen_candidate_2.index;
+                    break;
+                }
+            }
+        }
+        Logger::get("buildtruegenpair")
+            ->debug("Selected Particles: {} {}", genpair[0], genpair[1]);
+        if (genpair[0] == -1 || genpair[1] == -1) {
+            Logger::get("buildtruegenpair")
+                ->debug("no viable daughter particles found");
+            return genpair;
+        }
+        if (daughter_1_pdgid == daughter_2_pdgid) {
+            if (pts.at(genpair[0]) < pts.at(genpair[1])) {
+                std::swap(genpair[0], genpair[1]);
+            }
+        }
+
+        return genpair;
+    };
+    return df.Define(genpair, getTrueGenPair,
+                     {statusflags, status, pdgids, motherids, pts});
+}
+/// This function flags events, where a suitable particle pair is found.
+/// A pair is considered suitable, if a PairSelectionAlgo (like
+/// pairselection::mutau::PairSelectionAlgo) returns indices, that are
+/// not -1. Events, where any of the particle indices is -1 are vetoed
+/// by this filter.
 ///
 /// \param df The input dataframe
 /// \param flagname The name of the generated flag column
-/// \param pairname The name of the column, containing the particle indices
-/// index of the particle in the particle quantity vectors.
+/// \param pairname The name of the column, containing the particle
+/// indices index of the particle in the particle quantity vectors.
```

Comment:
I think this should be something like this.
```
\param pairname The name of the column, containing the 
indices of the particles in the particle quantity vectors.
```

Reply:
changes